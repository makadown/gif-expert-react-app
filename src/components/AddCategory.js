import React, { useState } from 'react'
import PropTypes from 'prop-types';

/**
 * Componente para capturar una nueva categoría
 * @param {agregarPapu} {setCategories} Referencia a Método del useState para agregar elemento a arreglo
 * @returns componente AddCategory
 */
export const AddCategory = ( { agregarPapu } ) => {
    const [inputValue, setInputValue] = useState('');
    const handleInputChange = (e) => {
        setInputValue(e.target.value);
    };

    const handleSubmit = (e) => {
        e.preventDefault();

        if (inputValue.trim().length > 2) {
            // aqui se agrega elemento al arreglo que esta en 
            // componente GifExpertApp mediante el metodo del useState.
            agregarPapu( cats => [inputValue, ...cats, ]);
            setInputValue('');
        }
    };

    return (
        <>
            <form onSubmit={ handleSubmit }>
                <input
                    type="text"
                    value={inputValue}
                    onChange={handleInputChange}
                />
            </form>
        </>
    )
}

// Candado para que se force a usar este parametro al usar este componente
AddCategory.propTypes = {
    agregarPapu: PropTypes.func.isRequired
}