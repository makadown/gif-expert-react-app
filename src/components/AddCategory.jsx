import { useState } from "react";
import PropTypes from 'prop-types';

export const AddCategory = ( { onAddCategory } ) => {

    const [inputValue, setInputValue] = useState('');

    const onInputChange = ({target}) => {
        setInputValue(target.value);
    }

    const onSubmit = (e) => {
        e.preventDefault();
        const cleanInput = inputValue.trim();
        
        if (cleanInput.length <= 1) return;        
        
        if ( onAddCategory( cleanInput ) ) {
            setInputValue('');
        } else {
            window.alert('Category already exists');
        }
    }
    

    return (
        <form onSubmit={ onSubmit }>
            <input type="text" placeholder="Search gifs" id="myCategory" 
                    name="myCategory" style={{width: '25%', height: '10px'}}
                    value={inputValue} onChange={ onInputChange }
                    />
        </form>
    )
}

AddCategory.propTypes = {
    onAddCategory: PropTypes.func.isRequired,
}
