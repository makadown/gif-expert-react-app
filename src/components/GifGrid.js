import React from 'react'

export const GifGrid = ({ category }) => {

    const getGifs = async() => {
        const palabraClave = 'Rick and morty'
                .replace(' ','+').replace(' ','+').replace(' ','+');
        const apiKey = 'api_key=3Ie0XE3iXNg8qBqu6WJO2J38EheOg33Z';
        const url = 'https://api.giphy.com/v1/gifs/search?q=' +
                palabraClave + '&limit=10&'+
                apiKey;
        const respuesta = await fetch(url);
        const { data } = await respuesta.json();
        console.log(data);
        const gifs = data.map( d => {
            return { 
                id: d.id,
                title: d.title,
                url: d.images?.downsized_medium.url
            };
        });
        console.log(gifs);
    };
    getGifs();
    return (
        <>
            <li> {category} </li>
        </>
    )
}
