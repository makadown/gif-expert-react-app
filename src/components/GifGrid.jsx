import { useEffect } from "react";
import { getGifs } from "../helpers/getGifs";

export const GifGrid = ( { category } ) => {
    
    // using useEffect I need to call the getGifs function
    // - Only once and not every time the component is rendered
    // - Only if the category changes
    useEffect( () => {
        getGifs( category );
    }, [ category ] );

    

    return (
        <>
            <h3>{category} </h3>
            <p>Hola mundo!</p>
        </>
    )
}