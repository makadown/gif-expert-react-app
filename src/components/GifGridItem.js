import React from 'react'

export const GifGridItem = ( img ) => {
    console.log('recibiendo en GifGridItem', img);
    return (
        <div>
            <img src={img.url} alt={ img.title }/>
            <p>{img.title}</p>
        </div>
    );
}
