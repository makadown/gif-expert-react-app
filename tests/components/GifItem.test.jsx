// Use React Testing Library to create unit test for GifItem and Evaluate snapshot
import {render} from '@testing-library/react';
import {GifItem} from '../../src/components/GifItem';

describe('Pruebas en GifItem', () => {
    const title = 'title';
    const url = 'https://url.com/img.png';

    test('should match the snapshot', () => {
        const {container} = render(<GifItem title={title} url={url}/>);
        expect(container).toMatchSnapshot();
    });

    test('should show the image with the given URL and ALT attribute', () => {
        const {container} = render(<GifItem title={title} url={url}/>);
        const img = container.querySelector('img');
        expect(img.src).toBe(url);
        expect(img.alt).toBe(title);
    });

    test('should show the title within the component', () => {
        const {container} = render(<GifItem title={title} url={url}/>);
        const p = container.querySelector('p');
        expect(p.textContent).toBe(title);
    })
});
