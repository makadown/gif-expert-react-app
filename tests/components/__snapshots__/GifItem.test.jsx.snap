// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifItem should match the snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="title"
      src="https://url.com/img.png"
    />
    <p>
      title
    </p>
  </div>
</div>
`;
